cmd/frontend/internal/pkg/search/query/types/check_test.go

```diff
@@ -144,3 +144,61 @@ func TestUnquoteString(t *testing.T) {
 		})
 	}
 }
+
+func TestCheck_fixup(t *testing.T) {
+	toValue := func(v *Value) value { return value{Not: v.Not(), Value: v.Value()} }
+	toTestValueMap := func(fields map[string][]*Value) map[string][]value {
+		m := make(map[string][]value, len(fields))
+		for f, vs := range fields {
+			m[f] = make([]value, len(vs))
+			for i, v := range vs {
+				m[f][i] = toValue(v)
+			}
+		}
+		return m
+	}
+
+	conf := Config{
+		FieldTypes: map[string]FieldType{
+			"": {
+				Literal: RegexpType,
+				Quoted:  StringType,
+			},
+		},
+		FieldAliases: map[string]string{},
+	}
+
+	tests := map[string]struct {
+		want    map[string][]value
+		wantErr *TypeError
+	}{
+		"foo(":         {want: map[string][]value{"": {{Value: regexp.MustCompile(`foo\(`)}}}},
```

Comment:
There is some redundancy in the `want` definition here that makes the test cases harder to read. Are you going to add tests cases for wantErr?

```suggestion
		"foo(":         {want: regexp.MustCompile(`foo\(`)},
```

Reply:
I rewrote the test to not copy/paste boiler plate from the other test. Now there is a test that only  calls the function that handles the fixup. This will make it easier to port the test over to the new query package when we get around to it.